pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -632,6 +632,59 @@ public static int hammingDistance(String input1, String input2) {
     return distance;
   }
 
+  /**
+   * Calculates the Levenshtein edit distance between two strings.
+   * The Levenshtein distance is the minimum number of single-character edits
+   * (insertions, deletions, or substitutions) needed to transform one string into another.
+   * This complements the existing hammingDistance function by handling strings of different lengths.
+   *
+   * @param input1 First string
+   * @param input2 Second string
+   * @return The Levenshtein distance between the two strings, or -1 if either input is null
+   */
+  @ScalarFunction
+  public static int levenshteinDistance(String input1, String input2) {
```

Comment:
Do you want to explicitly handle `null` input, or let the framework handles it and return `null` when any of the input is `null`?